EIPS/eip-network_upgrade_windows.md

```diff
@@ -0,0 +1,171 @@
+
+A proposal to define a limited number of annual time windows in which network
+upgrades (aka hard forks) should be performed within. Policies for scheduling
+network upgrades outside these windows are also described.
+
+## Abstract
+
+Four different weeks, spaced roughly evenly throughout the year, are targeted
+for network upgrades to be launched. Regular network upgrades should announce
+their intention to launch in a particular window early in their process and
+choose a block number four to six weeks prior to that window. If a network
+upgrade is cancelled then it would be rescheduled for the next window. Not all
+windows will be used. Priority upgrades outside the roadmap may be scheduled in
+the third week of any month, but such use is discouraged. Critical upgrades are
+scheduled as needed.
+
+## Motivation
+
+The aim of this EIP is to provide some level of regularity and predictability to
+the Ethereum network upgrade/hard fork process. This will allow service
+providers such as exchanges and node operators a predictable framework to
+schedule activities around. This also provides a framework to regularize the
+delivery of network upgrades.
+
+## Specification
+
+Scheduling is defined for three categories of network upgrades. First are
+`Roadmap` network upgrades that include deliberate protocol improvements. Next
+are `Priority` network updates, where there are technical reasons that
+necessitate a prompt protocol change but these reasons do not present a systemic
+risk to the protocol or the ecosystem. Finally, `Critical` network upgrades are
+to address issues that present a systemic risk to the protocol or the ecosystem.
+
+### Roadmap Network Upgrades
+
+Roadmap network upgrades are network upgrades that are deliberate and measured
+to improve the protocol and ecosystem. Historical examples are Homestead,
+Byzantium, and Constantinople.
+
+Roadmap network upgrades should be scheduled in one of four windows: the third
+week of January, the third week fo April, the third week of July or the third
+week of October. When initiating a network upgrade or early in the planning
+process a particular window should be targeted. A block number for the network
+upgrade should be chosen four to six weeks prior to the network upgrade window.
+Scheduling details such as whether this choice is made prior to or after testnet
+deployment are out of scope of this EIP.
+
+> **Note to reviewers:** The months and week chosen are to provide an initial
+> recommendation and are easily modifiable prior to final call. They thread the
+> needle between many third quarter and fourth quarter holidays.
+
+A roadmap upgrade typically should not occur in adjacent upgrade windows. Hence
+if a roadmap upgrade occurred in April then the July window would not be used
+for network upgrades.
+
+If a planned roadmap upgrade needs to be rescheduled then strong consideration
+should be given to rescheduling the upgrade for the next window in three months
+time.
```

Comment:
It seems like this is the reason for doing the above. Explicitly make that link if so

Reply:
Instead of specifying a 6 or more month cadence I added text stating the cadence should be a multiple of three months.

Reply:
I'm thinking that it could perhaps be stated more clearly as "The reason for leaving open the 3 month window after the proposed slot is to create space for recovery if things go wrong. For example, if an update is missed in a specific time window, it can be easily rescheduled for the following window without the pain of having to figure out multiple teams schedules in an emergency scenario"

Reply:
I agree, but I would slowly be moving into specifying how roadmap upgrades scheduling and cadence work, which should should be its own EIP.  It is still uncertain if we want 9, 6, or 3 month upgrades.  3 month upgrades would not permit empty slots.  at 6 and 9 the impact of a missed upgrade are different, at 9 a slip of one window still allows for an empty window.  But these discussions and specifications are better suited for a roadmap cadence EIP.